tests/testthat/test-tokens-word4.R

```diff
@@ -936,7 +894,7 @@ test_that("split_tags works", {
     txt2 <- c(d1 = "#quanteda #q-x #q_y #q100 #q")
     expect_identical(
         as.list(tokens(txt2, what = "word")),
-        list(d1 = c("#quanteda", "#q-x", "#q_y", "#q100", "#q"))
+        list(d1 = c("#quanteda", "#q", "-", "x", "#q_y", "#q100", "#q"))
```

Comment:
This really needs to be resolved, it's a flaw in "word4".

Reply:
They are different because of 

```
> stringi::stri_match_all_regex("#hash_tag", "#[\\p{L}\\p{N}]+#?")
[[1]]
     [,1]   
[1,] "#hash"

> stringi::stri_match_all_regex("#hash_tag", "#\\w+#?")
[[1]]
     [,1]       
[1,] "#hash_tag"
```
We can revert the default value for `pattern_hashtag` to the latter.